### PR TITLE
143 expand linexy constructors and slope information

### PR DESCRIPTION
**Pull request by bbean23**

### CODE REVIEW

File: opencsp/common/lib/geometry/LineXY.py [hunk 1]
```diff
@@ -188,7 +250,65 @@ def from_two_points(cls, Pxy1: Vxy, Pxy2: Vxy):
...
+        Parameters
+        ----------
+        rho : float
+            The right angle distance between the line and the origin (0,0). For
+            images, this will be the top-left corner of the image.
+        theta : float
+            The angle of the line, in radians, for the standard graphing
+            coordinate system. 0 is on the positive x-axis to the right, and the
+            angle increases counter-clockwise.
+        """
```
Comment (braden6521) [inline opencsp/common/lib/geometry/LineXY.py:271]: Can you give an example here or clarify. For example, if you had rho=1 and theta=0, would the line pass through x=+1 or x=-1?

**Review by braden6521 — APPROVED** [review]
Looks good! I left one suggestion for extra documentation. Feel free to take or leave it.

File: opencsp/common/lib/geometry/LineXY.py (hunk 1 above)
Comment (bbean23) [reply opencsp/common/lib/geometry/LineXY.py]: I'm glad you left a comment. I double-checked the method, and realized I had both a bug and a bad definition.

* Changed the definition of theta from "angle of the line" to "angle between the x-axis and the right angle distance vector".
* Removed the cast to an integer, which was causing numeric issues.
* Added unit tests for from_rho_theta() and from_location_angle().

**Review by braden6521 — APPROVED** [review]
Looks good!

**Pull request merged**